homeassistant/components/tesla_fleet/__init__.py

```diff
@@ -0,0 +1,170 @@
+"""Tesla Fleet integration."""
+
+import asyncio
+from typing import TYPE_CHECKING, Final
+
+import jwt
+from tesla_fleet_api import EnergySpecific, TeslaFleetApi, VehicleSpecific
+from tesla_fleet_api.const import Scope
+from tesla_fleet_api.exceptions import (
+    InvalidToken,
+    LoginRequired,
+    OAuthExpired,
+    TeslaFleetError,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_TOKEN, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.config_entry_oauth2_flow import (
+    OAuth2Session,
+    async_get_config_entry_implementation,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import DeviceInfo
+
+from .const import DOMAIN, MODELS
+from .coordinator import (
+    TeslaFleetEnergySiteInfoCoordinator,
+    TeslaFleetEnergySiteLiveCoordinator,
+    TeslaFleetVehicleDataCoordinator,
+)
+from .models import TeslaFleetData, TeslaFleetEnergyData, TeslaFleetVehicleData
+
+PLATFORMS: Final = [Platform.SENSOR]
+
+type TeslaFleetConfigEntry = ConfigEntry[TeslaFleetData]
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: TeslaFleetConfigEntry) -> bool:
+    """Set up TeslaFleet config."""
+
+    access_token = entry.data[CONF_TOKEN][CONF_ACCESS_TOKEN]
+    session = async_get_clientsession(hass)
+
+    token = jwt.decode(access_token, options={"verify_signature": False})
+    scopes = token["scp"]
+    region = token["ou_code"].lower()
+
+    # Setup OAuth refresh
+    implementation = await async_get_config_entry_implementation(hass, entry)
+    oauth_session = OAuth2Session(hass, entry, implementation)
+    refresh_lock = asyncio.Lock()
+
+    async def _refresh_token() -> str:
+        async with refresh_lock:
+            await oauth_session.async_ensure_token_valid()
+            token = oauth_session.token[CONF_ACCESS_TOKEN]
```

Comment:
I think if you change this to `token: str = oauth_session.token[CONF_ACCESS_TOKEN]` you can remove the type checking below

Reply:
Thanks!